decision_maker.go

```diff
@@ -393,36 +397,37 @@ func getValuesFiles(r *release) string {
 		fileList = append(fileList, r.SecretsFiles...)
 	}
 
-	if len(fileList) > 0 {
-		return " -f " + strings.Join(fileList, " -f ")
+	fileListArgs := []string{}
+	for _, file := range fileList {
+		fileListArgs = append(fileListArgs, "-f", file)
 	}
-	return ""
+	return fileListArgs
 }
 
 // getSetValues returns --set params to be used with helm install/upgrade commands
-func getSetValues(r *release) string {
-	result := ""
+func getSetValues(r *release) []string {
+	result := []string{}
 	for k, v := range r.Set {
-		result = result + " --set " + k + "=\"" + strings.Replace(v, ",", "\\,", -1) + "\""
+		result = append(result, "--set", k+"="+strings.Replace(v, ",", "\\,", -1)+"")
```

Comment:
This here seems to be different. The resulting string would be `k=something` instead of k="something". It would break if `v` have any space. Unless `v` is never supposed to have space in it.

Reply:
No, it does not break if you enter a space.

Reply:
Gotcha, just making sure.

Reply:
Refactors are hard and sometimes, small details will destroy the effort, just making sure that this particular ommision of the double quote was really intended and not a typo.